### PR TITLE
Mysql java connector proxy support

### PR DESCRIPTION
**Pull request by mattpgh**
With the recent addition of proxy support, I found it was missing proxy support for the mysql-connector. It also was missing puppet-archive support.

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @mattpgh, thanks for the PR. You touched a lot of files that aren't related to the change, please don't do this. They are templated and come from our [modulesync_config](https://github.com/voxpupuli/modulesync_config). Can you please remove them from the branch and rebase?

**Comment by mattpgh** [comment]
Sorry about that, just used the pdk as I normally would without thinking about it. 

Is this better now?

File: manifests/mysql_connector.pp [hunk 1]
```diff
@@ -5,34 +5,66 @@
   $format       = $jira::mysql_connector_format,
   $installdir   = $jira::mysql_connector_install,
   $download_url = $jira::mysql_connector_url,
+  $deploy_module= $jira::deploy_module,
```
Comment (bastelfreak) [inline manifests/mysql_connector.pp]: there should be a whitespace before the =

File: manifests/mysql_connector.pp [hunk 2]
```diff
@@ -5,34 +5,66 @@
...
 
-  if ! defined(File[$installdir]) {
-    file { $installdir:
-      ensure => 'directory',
-      owner  => root,
-      group  => root,
-      before => Staging::File[$file],
+->  case $deploy_module {
+    'staging': {
+      require ::staging
```
Comment (bastelfreak) [inline manifests/mysql_connector.pp]: You don't need to add :: to the class scope

**Comment by bastelfreak** [comment]
Thanks. Can you take a look at the failing travis jobs? The complain about hard tabs / wrong indentation

**Comment by ekohl** [comment]
The current version does appear to pass proxy params and uses the archive module. Please open a new PR if you're still missing anything.

**Pull request closed without merging**